xarray/conventions.py

```diff
@@ -350,6 +350,25 @@ def stackable(dim):
         drop_variables = []
     drop_variables = set(drop_variables)
 
+    # Time bounds coordinates might miss the decoding attributes
+    # https://github.com/pydata/xarray/issues/2565
+    if decode_times:
+        # We list all time variables with bounds
+        to_update = dict()
+        for v in variables.values():
+            attrs = v.attrs
+            if ('units' in attrs and 'since' in attrs['units'] and
+                    'bounds' in attrs):
+                new_attrs = {'units': attrs['units']}
+                if 'calendar' in attrs:
+                    new_attrs['calendar'] = attrs['calendar']
+                to_update[attrs['bounds']] = new_attrs
+        # For all bound variables, update their time attribute if not present
+        for k, new_attrs in to_update.items():
```

Comment:
I think we could update the attributes of the potential bounds variables immediately within the `for v in variables.values()` loop rather than use another loop here (this would remove the need for the `to_update` dictionary).